include/verilated_vcd_c.cpp

```diff
@@ -804,6 +874,266 @@ void VerilatedVcd::fullBusX(vluint32_t code, int bits) {
+                                          int wholeWords);
+template void VerilatedVcd::fullArray<17>(vluint32_t* oldp, const vluint32_t* newval,
+                                          int wholeWords);
+template void VerilatedVcd::fullArray<18>(vluint32_t* oldp, const vluint32_t* newval,
+                                          int wholeWords);
+template void VerilatedVcd::fullArray<19>(vluint32_t* oldp, const vluint32_t* newval,
+                                          int wholeWords);
+template void VerilatedVcd::fullArray<20>(vluint32_t* oldp, const vluint32_t* newval,
+                                          int wholeWords);
+template void VerilatedVcd::fullArray<21>(vluint32_t* oldp, const vluint32_t* newval,
+                                          int wholeWords);
+template void VerilatedVcd::fullArray<22>(vluint32_t* oldp, const vluint32_t* newval,
+                                          int wholeWords);
+template void VerilatedVcd::fullArray<23>(vluint32_t* oldp, const vluint32_t* newval,
+                                          int wholeWords);
+template void VerilatedVcd::fullArray<24>(vluint32_t* oldp, const vluint32_t* newval,
+                                          int wholeWords);
+template void VerilatedVcd::fullArray<25>(vluint32_t* oldp, const vluint32_t* newval,
+                                          int wholeWords);
+template void VerilatedVcd::fullArray<26>(vluint32_t* oldp, const vluint32_t* newval,
+                                          int wholeWords);
+template void VerilatedVcd::fullArray<27>(vluint32_t* oldp, const vluint32_t* newval,
+                                          int wholeWords);
+template void VerilatedVcd::fullArray<28>(vluint32_t* oldp, const vluint32_t* newval,
+                                          int wholeWords);
+template void VerilatedVcd::fullArray<29>(vluint32_t* oldp, const vluint32_t* newval,
+                                          int wholeWords);
+template void VerilatedVcd::fullArray<30>(vluint32_t* oldp, const vluint32_t* newval,
+                                          int wholeWords);
+template void VerilatedVcd::fullArray<31>(vluint32_t* oldp, const vluint32_t* newval,
+                                          int wholeWords);
+
+void VerilatedVcd::fullFloat(vluint32_t* oldp, float newval) {
+    // cppcheck-suppress invalidPointerCast
+    *reinterpret_cast<float*>(oldp) = newval;
+    char* wp = m_writep;
+    // Buffer can't overflow before sprintf; we sized during declaration
+    sprintf(wp, "r%.16g", static_cast<double>(newval));
+    wp += strlen(wp);
+    finishLine(oldp, wp);
+}
+
+void VerilatedVcd::fullDouble(vluint32_t* oldp, double newval) {
+    // cppcheck-suppress invalidPointerCast
+    *reinterpret_cast<double*>(oldp) = newval;
+    char* wp = m_writep;
+    // Buffer can't overflow before sprintf; we sized during declaration
+    sprintf(wp, "r%.16g", newval);
```

Comment:
This will probably be slow for someone, but I don't know of anything faster that's in the standard library until C++11.

Reply:
Yes, thats just how it has always been and I wan't very worried about doubles. Curious, what would that be in C++11?

Reply:
I was thinking std::to_string(d), but a quick benchmark says it's slightly slower than sprintf which is highly disappointing, maybe is the string overhead.  I guess if this becomes a problem we have to roll our own.

Reply:
BTW looking at sprintf it always promotes floats to doubles, so we might as well have fullFloat just upcast and call fullDouble.

Reply:
Reference for future: https://github.com/miloyip/dtoa-benchmark

Reply:
`fullFloat` and `fullDouble` (as all `full*` functions) also do the write back to the old value buffer, so can't quite call `fullDouble` instead of `fullFloat` because `sizeof(float) != sizeof(double)`.

Reply:
<<=1 that is.